server/users/users.test.js

```diff
@@ -21,16 +21,43 @@ describe("/api/users tests", () => {
     await User.remove({});
   });
 
-  it("should post and get a user", async () => {
-    const postResponse = await request(app)
-      .post("/api/users")
-      .send({ name: "John Doe", course: "SOFTENG 700" });
+  it("should be able to create a new user", async () => {
+    const dateNow = new Date();
+
+    const postResponse = await request(app).post("/api/users").send({
+      name: "John Doe",
+      passwordHash: "aMoreSecureHashThanThisPlease",
+      email: "john@doe.com",
+      position: "Course Cordinator",
+      department: "Engineering",
+      institution: "The University Of Auckland",
+      createdAt: dateNow,
+      username: "jdoe",
+    });
+    expect(postResponse.status).toBe(200);
+  });
+
+  it("should be able to authenticate a user", async () => {
+    const dateNow = new Date();
+
+    const postResponse = await request(app).post("/api/users").send({
+      name: "John Doe",
+      passwordHash: "aMoreSecureHashThanThisPlease",
+      email: "john@doe.com",
+      position: "Course Cordinator",
+      department: "Engineering",
+      institution: "The University Of Auckland",
+      createdAt: dateNow,
+      username: "jdoe",
+    });
     expect(postResponse.status).toBe(200);
 
-    const getResponse = await request(app).get("/api/users");
-    expect(getResponse.status).toBe(200);
-    expect(getResponse.body).toEqual([
-      expect.objectContaining({ name: "John Doe", course: "SOFTENG 700" }),
-    ]);
+    const authResponse = await request(app)
+      .post("/api/users/authenticate/jdoe")
+      .send({
+        passwordHash: "aMoreSecureHashThanThisPlease",
+      });
+    expect(authResponse.status).toBe(200);
+    expect(authResponse.body.username).toEqual("jdoe");
   });
 });
```

Comment:
Should there be a test for ensuring that an incorrect password does not result in successful auth?

Reply:
knew I was forgetting something! thanks